app/src/test/java/com/android/streetworkapp/model/park/ParkRatingTest.kt

```diff
@@ -0,0 +1,125 @@
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.never
+
+class ParkRatingTest {
+  private lateinit var parkRepository: ParkRepository
+  private lateinit var parkViewModel: ParkViewModel
+  private val testDispatcher = StandardTestDispatcher()
+
+  @OptIn(ExperimentalCoroutinesApi::class)
+  @Before
+  fun setup() {
+    Dispatchers.setMain(testDispatcher)
+    parkRepository = Mockito.mock(ParkRepository::class.java)
+    parkViewModel = ParkViewModel(parkRepository)
+  }
+
+  @Test
+  fun handlerCorrectlyCallingAddRatingForStarRatingValues() = runTest {
+    // Instantiate basic (but correct) park and user :
+    val emptyPark =
+        Park("pid", "name", ParkLocation(0.0, 0.0, ""), "", 1f, 1, 10, 0, emptyList(), emptyList())
+
+    val user = User("uid", "username", "email", 0, emptyList())
+
+    val starRatingValues = listOf(1, 2, 3, 4, 5, 0, -1, 100)
+
+    // Any value of starRating different from 1 to 5 should not call addRating in handleRating :
+    for (starRating in starRatingValues) {
+      handleRating(null, emptyPark, user, starRating, parkViewModel)
+      testDispatcher.scheduler.advanceUntilIdle()
+
+      if (starRating in 1..5) {
+        Mockito.verify(parkRepository, Mockito.times(1))
+            .addRating(emptyPark.pid, user.uid, starRating.toFloat())
+      } else {
+        Mockito.verify(parkRepository, never())
+            .addRating(emptyPark.pid, user.uid, starRating.toFloat())
+      }
+    }
+  }
+
+  @Test
+  fun handlerCorrectlyCallingAddRatingForUserValues() = runTest {
+    // Instantiate basic (but correct) park and user :
+    val emptyPark =
+        Park("pid", "name", ParkLocation(0.0, 0.0, ""), "", 1f, 1, 10, 0, emptyList(), emptyList())
+
+    val user1 = User("uid1", "username1", "email1", 0, emptyList())
+    val user2 = User("uid2", "username2", "email2", 0, emptyList())
+    val user3 = User("uid3", "username3", "email3", 0, emptyList())
+
+    val userValues = listOf(null, user1, user2, null, user3)
+    val starRating = 4
+
+    // Any value of starRating different from 1 to 5 should not call addRating in handleRating :
+    for (user in userValues) {
+      handleRating(null, emptyPark, user, starRating, parkViewModel)
+      testDispatcher.scheduler.advanceUntilIdle()
+
+      if (user != null) {
+        Mockito.verify(parkRepository, Mockito.times(1))
+            .addRating(emptyPark.pid, user.uid, starRating.toFloat())
+      } else {
+        Mockito.verify(parkRepository, never())
+            .addRating(eq(emptyPark.pid), any(), eq(starRating.toFloat()))
+      }
+      Mockito.reset(parkRepository)
+    }
+  }
+
+  @Test
+  fun handlerCorrectlyCallingAddRatingForParkValues() = runTest {
+    val starRating = 1
+    val user = User("uid", "username", "email", 0, emptyList())
+
+    // Testing for park with no rating and park with rating :
+    val emptyPark =
```

Comment:
Here aswell all the empty park are the same, it can be init in a @before` setup function`